lib/big_query/base/timestamp_group.ex

```diff
@@ -0,0 +1,48 @@
+defmodule BigQuery.Base.TimestampGroup do
+  import BigQuery.Base.Query
+  import BigQuery.Base.Timestamp,
+    only: [timestamp_intervals: 3, timestamp_params: 2]
+
+  def group_query(tbl, where_sql, params, interval, grouping) do
+    params
+    |> timestamp_params(interval)
+    |> group_params(grouping)
+    |> query(group_sql(tbl, where_sql, grouping))
+  end
+
+  def group_sql(tbl, where_sql, grouping) do
+    """
+    WITH
+      intervals AS (#{timestamp_intervals(tbl, where_sql, grouping.fkey)}),
+      top_groups AS (#{top_groups("intervals", grouping.fkey)})
+    SELECT
+      time,
+      SUM(count) as count,
+      IF(rank IS NULL, NULL, ANY_VALUE(#{grouping.display})) AS display,
```

Comment:
That query is doing a `GROUP BY`, and rather than also grouping by the `geonames.city_name` column, it's slightly faster/cheaper just to grab any value out of that field.  (They're all going to be the same, since I'm already grouping by rank).

https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#any_value